exchange/utils.go

```diff
@@ -700,3 +701,39 @@ func getExtBidAdjustmentFactors(requestExt *openrtb_ext.ExtRequest) map[string]f
 	}
 	return bidAdjustmentFactors
 }
+
+func buildXPrebidHeader(bidRequest *openrtb2.BidRequest, version string) string {
+	req := &openrtb_ext.RequestWrapper{BidRequest: bidRequest}
+
+	sb := &strings.Builder{}
+	writeNameVersionRecord(sb, "pbs-go", version)
+
+	if reqExt, err := req.GetRequestExt(); err == nil && reqExt != nil {
+		if prebidExt := reqExt.GetPrebid(); prebidExt != nil {
+			if channel := prebidExt.Channel; channel != nil {
+				writeNameVersionRecord(sb, channel.Name, channel.Version)
+			}
+		}
+	}
+	if appExt, err := req.GetAppExt(); err == nil && appExt != nil {
+		if prebidExt := appExt.GetPrebid(); prebidExt != nil {
+			writeNameVersionRecord(sb, prebidExt.Source, prebidExt.Version)
+		}
+	}
```

Comment:
What if we have valid values in both `reqExt` and `appExt`? In other words, what if all conditions of `if` statements found in lines 711, 712, and 713 as well as 718 and 719 are all true? Do we want `buildXPrebidHeader` to have the ability to return something like:
```
"pbs-go/test-version,pbjs/test-pbjs-version,prebid-mobile/test-prebid-mobile-version"
                               ^                              ^---- mobile version
                               |----------channel version
```
If we don't want this scenario, should we maybe use a `return` statement if valid `channel.Name` and `channel.Version` are parsed?
```
705   func buildXPrebidHeader(bidRequest *openrtb2.BidRequest, version string) string {
706       req := &openrtb_ext.RequestWrapper{BidRequest: bidRequest}
707  
708       sb := &strings.Builder{}
709       writeNameVersionRecord(sb, "pbs-go", version)
710  
711       if reqExt, err := req.GetRequestExt(); err == nil && reqExt != nil {
712           if prebidExt := reqExt.GetPrebid(); prebidExt != nil {
713               if channel := prebidExt.Channel; channel != nil {
714                   writeNameVersionRecord(sb, channel.Name, channel.Version)
    +                 return sb.String()
715               }
716           }
717       }
718       if appExt, err := req.GetAppExt(); err == nil && appExt != nil {
719           if prebidExt := appExt.GetPrebid(); prebidExt != nil {
720               writeNameVersionRecord(sb, prebidExt.Source, prebidExt.Version)
721           }
722       }
723       return sb.String()
724   }
exchange/utils.go
```

Reply:
Please disregard my comment above I've been told it's fine to have both `reqExt` and `appExt` values.

Reply:
I think it's fine. I mean, it doesn't really make sense for both to be provided. This may never actually happen in the wild, but if it does - I don't personally see an issue with a comma separated list of 3 items. The bidder can parse out what they like.